### PR TITLE
Re-add libnavigation-core, examples and UI testing lib

### PR DESCRIPTION
**Pull request by LukasPaczos**
This PR:
- Brings in all the progress that's been done on the `libnavigation-core`, all modules and tests that are part of it and the new `MapboxNavigation` controller and entry point class.
- Brings the UI testing library, sets it up on CI and adds all missing examples.
- Fixes trip service activity's notification interaction.

### CODE REVIEW

**Comment by LukasPaczos** [comment]
Bintray repo for the `core` module is available in https://bintray.com/mapbox/mapbox/com.mapbox.navigation%3Acore. I'm going to setup the release process as soon as this is merged.

File: circle.yml [hunk 1]
```diff
@@ -139,6 +140,59 @@ jobs:
...
+          command: ./gradlew examples:assembleDebug
+      - run:
+          name: Build Instrumentation Test APK
+          command: ./gradlew examples:assembleAndroidTest
+      - run:
+          name: Log in to Google Cloud Platform
+          shell: /bin/bash -euo pipefail
+          command: |
+            echo "${GCLOUD_SERVICE_ACCOUNT_JSON}" > secret.json
+            gcloud auth activate-service-account --key-file secret.json --project driver-efb37
```
Comment (Guardiola31337) [inline circle.yml]:
```suggestion
            gcloud auth activate-service-account --key-file secret.json --project mapbox-navigation-android
```

https://github.com/mapbox/mapbox-navigation-android/blob/b5971b9b6f469827f3486471606f7fa7560a4270/circle.yml#L73

File: examples/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -16,11 +16,14 @@
     <string name="title_onboard_router_java" translatable="false">Onboard Router Java</string>
     <string name="description_onboard_router_java" translatable="false">Shows routes using onboard routing api in Java code.</string>
 
-    <string name="title_trip_service_kotlin">Trip Service Kotlin</string>
-    <string name="description_trip_service_kotlin">Shows how to use the Service to push notifications in Kotlin code.</string>
+    <string name="title_trip_service_kotlin" translatable="false">Trip Service Kotlin</string>
```
Comment (Guardiola31337) [inline examples/src/main/res/values/strings.xml]: Why `translatable="false"`? We currently have the test app project set up in Transifex and translators are contributing to it, we should do the same for `examples` module.

File: libnavigation-core/src/main/java/com/mapbox/navigation/MapboxNavigation.kt [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+            MapboxNavigationModuleType.OffboardRouter -> arrayOf(
+                String::class.java to (accessToken
+                    ?: throw RuntimeException("You need to provide an access in order to use the default OffboardRouter.")),
+                Context::class.java to context
+            )
+            MapboxNavigationModuleType.OnboardRouter -> arrayOf()
+            MapboxNavigationModuleType.DirectionsSession -> throw NotImplementedError() // going to be removed when next base version
+            MapboxNavigationModuleType.TripNotification -> arrayOf(
+                Context::class.java to context.applicationContext,
+                NavigationOptions::class.java to NavigationOptions.Builder(
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/MapboxNavigation.kt]: 👀 These options are currently hardcoded

File: libnavigation-core/src/main/java/com/mapbox/navigation/MapboxNavigation.kt [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+                    ::paramsProvider
+                )
+            )
+            MapboxNavigationModuleType.OffboardRouter -> arrayOf(
+                String::class.java to (accessToken
+                    ?: throw RuntimeException("You need to provide an access in order to use the default OffboardRouter.")),
+                Context::class.java to context
+            )
+            MapboxNavigationModuleType.OnboardRouter -> arrayOf()
+            MapboxNavigationModuleType.DirectionsSession -> throw NotImplementedError() // going to be removed when next base version
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/MapboxNavigation.kt]: `TODO` Same for some other modules below 👇

File: libnavigation-core/src/main/java/com/mapbox/navigation/trip/session/MapboxTripSession.kt [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+                isOffRoute = status.offRoute
+                delay(STATUS_POLLING_INTERVAL)
+            }
+        }
+    }
+
+    private suspend fun navigatorPolling(): TripStatus =
+        withContext(ioJobController.scope.coroutineContext) {
+            val date = Date()
+            date.time = date.time + 1500
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/trip/session/MapboxTripSession.kt]: `TODO` `delay` is currently hardcoded

File: examples/src/main/res/values/strings.xml (hunk 2 above)
Comment (LukasPaczos) [reply examples/src/main/res/values/strings.xml]: Why do we need translations for an example/debug project? It looks like we could direct that effort somewhere else.

File: circle.yml (hunk 1 above)
Comment (LukasPaczos) [reply circle.yml]: nice catch!

File: libnavigation-core/src/main/java/com/mapbox/navigation/MapboxNavigation.kt (hunk 4 above)
Comment (LukasPaczos) [reply libnavigation-core/src/main/java/com/mapbox/navigation/MapboxNavigation.kt]: https://github.com/mapbox/mapbox-base-android/pull/23

**Pull request merged**